internal/setup/setup.go

```diff
@@ -77,14 +82,29 @@ const (
 	UntilMutate PathUntil = "mutate"
 )
 
+type GenerateKind string
+
+const (
+	// There are only two possible values of "generate" as of now:
+	//   "" (empty)
+	//   "manifest"
+	// Empty value signifies that the path is not a GeneratePath.
+	GenerateNone     GenerateKind = ""
+	GenerateManifest GenerateKind = "manifest"
+)
+
 type PathInfo struct {
+	// Disable ==.
+	_ [0]func()
```

Comment:
I wanted to make it more explicit that it was not comparable and that you had to use `SameContent` to prevent someone changing the type of `PathInfo.Arch` in the future and suddenly the struct is comparable. Maybe we need a better comment `Disable ==, PathInfo.SameContent should be used to compare this struct`, thoughts?

Reply:
Changed in https://github.com/canonical/chisel/pull/143/commits/58531e9dccfef4f476e08647f2ebde51eed7979d.